openfisca_web_api/loader/entities.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals, print_function, division, absolute_import
+from openfisca_core.commons import to_unicode
+
+
+def build_entities(tax_benefit_system):
+    entities = {
+        entity.key: build_entity(entity)
+        for entity in tax_benefit_system.entities
+        }
+    return entities
+
+
+def build_entity(entity):
+
+    entity_formated = {
+        'plural': entity.plural,
+        'description': to_unicode(entity.doc)
+        }
+    if hasattr(entity, 'roles'):
+        entity_formated['roles'] = \
+            {
+            role.key: build_role(role)
+            for role in entity.roles
+            }
+    return entity_formated
+
+
+def build_role(role):
+    role_formated = {
+        'plural': role.plural,
+        'description': role.doc
+        }
+
+    if role.max:
+        role_formated['max'] = role.max
+    if role.subroles:
+        role_formated['max'] = len(role.subroles)
+
+    role_formated['mandatory'] = True if role_formated.get('max') else False
```

Comment:
No, there is no good reason to assume than a role with a max is a mandatory role. 

Right now, we don't handle mandatory role in OpenFisca Core. I see two solutions to deal with that:
1. Always set `role_formated['mandatory']` to `False` (the Web API would be "anticipating" the feature)
1. Open another PR in Core introducing this `mandatory` attribute in Python


1 allows for a quick delivery, but leads to a relatively inconsistent state of the codebase.

Reply:
I implemented 1 and opened #721 to follow 2